pkg/destination.go

```diff
@@ -17,17 +18,19 @@ type MongoDBDestination struct {
 	QueryTimeout            time.Duration
 	DatabaseName            string
 	CollectionName          string
+	UniqueIDFieldName       string
```

Comment:
This is pretty prescriptive, I think what would be better would be an array of `mongo.IndexModel` and each time it runs it creates all those indexes. Mongo's index stuff is idempotent and that would be much more flexible.

Reply:
I'll give that a go. 

I did it the way that it is now, because it was easy to build a filter that way: https://github.com/catalystsquad/data-mover-destination-mongodb/pull/3/files#diff-db75893d73fd1341dccadf924e943c8a12baeacb6bff4034b209865437abecb8R87

Reply:
I added your suggestion, and also added a field for specifying which data fields to use in the upsert filter. That seems to give a lot of flexibility. Just requires more input from the user.